### PR TITLE
Summernote busy spinner integration

### PR DESCRIPTION
**Pull request by klopfdreh**
https://github.com/l0rdn1kk0n/wicket-bootstrap/issues/537

@martin-g If you have the time - would you be so kind and review this? Thanks!

### CODE REVIEW

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteConfig.java [hunk 1]
```diff
@@ -33,6 +33,7 @@
     private static final IKey<Boolean> Force = newKey("force", null);
     private static final IKey<Integer> MaxFileSize = newKey("maxFilesize", 2097152);
     private static final IKey<String> ImageUploadCallbackUrl = newKey("imageUploadUrl", null);
+    private static final IKey<Integer> OverlayTimeout = newKey("overlayTimeout",2000);
```
Comment (jzaefferer) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteConfig.java]: Missing space after comma

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Singleton instance of this reference
+     */
+    private static final SummernoteEditorOverlayCssReference INSTANCE = new SummernoteEditorOverlayCssReference();
+
+    /**
+     * @return the single instance of the resource reference
+     */
+    public static SummernoteEditorOverlayCssReference instance() {
+	return INSTANCE;
```
Comment (jzaefferer) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java]: Missing/bad(?) indent, also below

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/js/summernote_init.js [hunk 3]
```diff
@@ -13,7 +13,17 @@ $(function() {
...
-            $(files).each(function(){
+        	var files = $(files);
+        	var filesSize = files.length;
+        	var overlay;
+        	
+        	// Show Overlay
+        	var overlayTimeout = setTimeout(function(){
+        		overlay = $("<div class='summernoteOverlay'></div>").appendTo("body");
+        		new Spinner({color:'#fff'}).spin(overlay[0]);
+        	},summernoteConfig.overlayTimeout);
```
Comment (jzaefferer) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/js/summernote_init.js]: Missing space after comma

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/js/summernote_init.js [hunk 4]
```diff
@@ -13,7 +13,17 @@ $(function() {
     var summernoteConfigDefault = {
         toolbar : toolbar,
         onImageUpload : function(files) {
-            $(files).each(function(){
+        	var files = $(files);
+        	var filesSize = files.length;
+        	var overlay;
+        	
+        	// Show Overlay
+        	var overlayTimeout = setTimeout(function(){
```
Comment (jzaefferer) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/js/summernote_init.js]: Missing space before `{`, more below

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java (hunk 2 above)
Comment (klopfdreh) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java]: Yep - it seems to be my formatter - my bad. 

@martin-g: Maybe we should add a formatter for eclipse (which also can be used for Intellij IDEA) to wicket bootstrap.

**Comment by klopfdreh** [comment]
@jzaefferer: Thanks for reviewing! :smile:

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java (hunk 2 above)
Comment (martin-g) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java]: @klopfdreh So you offer me your broken code formating config? No, thanks! :-)
Don't worry. I fix those after merging your PRs.

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/css/summernote.css [hunk 5]
```diff
@@ -0,0 +1,16 @@
+.summernoteOverlay{
+	z-index:16777271;
```
Comment (martin-g) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/css/summernote.css]: Almost reached Integer.MAX_VALUE here :-)

**Comment by martin-g** [comment]
:+1:

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/css/summernote.css [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+	width:100%;
+	position: fixed;
+	top:0;
+	bottom:0;
+	left:0;
+	right:0;
+	-ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=50)";
+	filter: alpha(opacity=50);
+	-moz-opacity: 0.5;
+	-khtml-opacity: 0.5;
```
Comment (martin-g) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/css/summernote.css]: According to http://caniuse.com/#search=opacity it seems we don't need `-moz` and `-khtml`.
May I nuke 'em?

Comment (martin-g) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/css/summernote.css]: Tested with Firefox and everything is OK.
Please confirm for Safari.

Comment (klopfdreh) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/css/summernote.css]: filter is ok for Safari

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/css/summernote.css (hunk 5 above)
Comment (klopfdreh) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/css/summernote.css]: Hehe Jep Safari 3 max value.. :smile:

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java (hunk 2 above)
Comment (klopfdreh) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java]: @martin-g: no no no no no no and no! :laughing: I just want to format everything right so you don't have to look at each class / fix it - but I want to do it with IDE support!

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/css/summernote.css (hunk 6 above)
Comment (klopfdreh) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/css/summernote.css]: Oh wait!! Use opacity: 0.5 without moz prefix! That should do it!

Comment (klopfdreh) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/css/summernote.css]: Ok you did it already - everything is fine!!

File: bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java (hunk 2 above)
Comment (martin-g) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java]: Do you talk about Wicket's Eclipse .jar?
Or you are willing to create a new .jar with the rules this project uses?

If it is the latter then I'm fine.
@l0rdn1kk0n prefers the current code style and as a owner of the project we should respect that.

Comment (klopfdreh) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java]: I just mean a formatter XML for eclipse with the rules of @l0rdn1kk0n

Comment (martin-g) [inline bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditorOverlayCssReference.java]: OK, then :+1: !

**Pull request merged**